build-restler.py struggles with spaces in folder name.

The build-restler.py struggles with spaces in folder names, for example:

Bad:
```
 C:\Users\User\Documents\Coding Project>python build-restler.py --dest_dir bin
 
 For switch syntax, type "MSBuild -help"
Publishing project C:\Users\User\Documents\Coding Project\restler-fuzzer\src\ResultsAnalyzer\ResultsAnalyzer.fsproj to output dir C:\Users\User\Documents\Coding Project\restler-fuzzer\restler_bin\resultsAnalyzer
MSBUILD : error MSB1008: Only one project can be specified.
Switch: Project\restler-fuzzer\src\ResultsAnalyzer\ResultsAnalyzer.fsproj
```

Ok:
```
 C:\Users\User\Documents\Coding>python build-restler.py --dest_dir bin 
 
 Generating a new RESTler binary drop...
Publishing dotnet core apps...
Publishing project C:\Users\User\Documents\Coding\restler-fuzzer-7.2.0\src\compiler\Restler.CompilerExe\Restler.CompilerExe.fsproj to output dir bin\compiler
  Determining projects to restore...
  Restored C:\Users\User\Documents\Coding\restler-fuzzer-7.2.0\src\compiler\Restler.Compiler\Restler.Compiler.fsproj (in 8.81 sec).
  Restored C:\Users\User\Documents\Coding\restler-fuzzer-7.2.0\src\compiler\Restler.CompilerExe\Restler.CompilerExe.fsproj (in 8.81 sec).
```

As you can see in the "Switch" it cuts off  C:\Users\User\Documents\Coding and just goes with just Project.

--- a/build-restler.py
+++ b/build-restler.py
@@ -160,12 +160,12 @@ def publish_dotnet_apps(dirs, configuration):
         proj_file_path = dotnetcore_projects[target_dir_name]
         print(f"Publishing project {proj_file_path} to output dir {proj_output_dir}")
 
-        output = subprocess.run(f"dotnet restore {proj_file_path} --use-lock-file --locked-mode --force", shell=True, stderr=subprocess.PIPE)
+        output = subprocess.run(f"dotnet restore \"{proj_file_path}\" --use-lock-file --locked-mode --force", shell=True, stderr=subprocess.PIPE)
         if output.stderr:
             print("Build failed!")
             print(str(output.stderr))
             sys.exit(-1)
-        output = subprocess.run(f"dotnet publish {proj_file_path} --no-restore -o {proj_output_dir} -c {configuration} -f netcoreapp5.0", shell=True, stderr=subprocess.PIPE)
+        output = subprocess.run(f"dotnet publish \"{proj_file_path}\" --no-restore -o \"{proj_output_dir}\" -c {configuration} -f netcoreapp5.0", shell=True, stderr=subprocess.PIPE)
         if output.stderr:
             print("Build failed!")
             print(str(output.stderr))

--- a/restler-quick-start.py
+++ b/restler-quick-start.py
@@ -38,7 +38,7 @@ def compile_spec(api_spec_path, restler_dll_path):
         os.makedirs(RESTLER_TEMP_DIR)
 
     with usedir(RESTLER_TEMP_DIR):
-        subprocess.run(f'dotnet {restler_dll_path} compile --api_spec {api_spec_path}', shell=True)
+        subprocess.run(f'dotnet \"{restler_dll_path}\" compile --api_spec \"{api_spec_path}\"', shell=True)
 
 def test_spec(ip, port, host, use_ssl, restler_dll_path):
     """ Runs RESTler's test mode on a specified Compile directory
@@ -60,10 +60,12 @@ def test_spec(ip, port, host, use_ssl, restler_dll_path):
     """
     with usedir(RESTLER_TEMP_DIR):
         compile_dir = Path(f'Compile')
-
+        grammar_file_path = compile_dir.joinpath('grammar.py')
+        dictionary_file_path = compile_dir.joinpath('dict.json')
+        settings_file_path = compile_dir.joinpath('engine_settings.json')
         command = (
-            f"dotnet {restler_dll_path} test --grammar_file {compile_dir.joinpath('grammar.py')} --dictionary_file {compile_dir.joinpath('dict.json')}"
-            f" --settings {compile_dir.joinpath('engine_settings.json')}"
+            f"dotnet \"{restler_dll_path}\" test --grammar_file \"{grammar_file_path}\" --dictionary_file \"{dictionary_file_path}\""
+            f" --settings \"{settings_file_path}\""
         )
         if not use_ssl:
             command = f"{command} --no_ssl"

--- a/src/driver/Program.fs
+++ b/src/driver/Program.fs
@@ -232,7 +232,7 @@ module Fuzz =
              | Some r -> sprintf "--path_regex %s" r
              | None -> "")
             (if String.IsNullOrWhiteSpace parameters.settingsFilePath then ""
-             else sprintf "--settings %s" parameters.settingsFilePath)
+             else sprintf "--settings \"%s\"" parameters.settingsFilePath)
 
             // Checkers
             (if parameters.checkerOptions.Length > 0 then
@@ -294,7 +294,7 @@ module Fuzz =
                     usage()
                 | Some (pythonFilePath, versionString) ->
                     printfn "Using python: '%s' (%s)" pythonFilePath (versionString.Trim())
-                    pythonFilePath, (sprintf "-B %s %s" Paths.RestlerPyPath restlerParameterCmdLine)
+                    pythonFilePath, (sprintf "-B \"%s\" %s" Paths.RestlerPyPath restlerParameterCmdLine)
 
         let! result =
             startProcessAsync

